### PR TITLE
Add try-with-resources to HttpWaitStrategyTest

### PR DESCRIPTION
**Pull request by eddumelendez**

### CODE REVIEW

File: core/src/test/java/org/testcontainers/junit/wait/strategy/AbstractWaitStrategyTest.java [hunk 1]
```diff
@@ -84,7 +84,11 @@ protected GenericContainer<?> startContainerWithCommand(
      * @param shellCommand the shell command to execute
      */
     protected void waitUntilReadyAndSucceed(String shellCommand) {
-        waitUntilReadyAndSucceed(startContainerWithCommand(shellCommand));
+        try (
+        GenericContainer<?> container = startContainerWithCommand(shellCommand)
+        ) {
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/wait/strategy/AbstractWaitStrategyTest.java:89]: is this correctly formatted? o_O I have never seen Prettier doing it like this 😅

Comment (eddumelendez) [reply core/src/test/java/org/testcontainers/junit/wait/strategy/AbstractWaitStrategyTest.java:89]: build fails because of that 🤔

Comment (bsideup) [reply core/src/test/java/org/testcontainers/junit/wait/strategy/AbstractWaitStrategyTest.java]: ahhh... yay :D

File: core/src/test/java/org/testcontainers/junit/wait/strategy/HttpWaitStrategyTest.java [hunk 2]
```diff
@@ -45,16 +45,19 @@ public void testWaitUntilReadyWithSuccess() {
     public void testWaitUntilReadyWithSuccessWithCustomHeaders() {
         HashMap<String, String> headers = new HashMap<>();
         headers.put("baz", "boo");
-        GenericContainer container = startContainerWithCommand(
-            createShellCommand("200 OK", GOOD_RESPONSE_BODY),
-            createHttpWaitStrategy(ready).withHeader("foo", "bar").withHeaders(headers)
-        );
-        waitUntilReadyAndSucceed(container);
+        try (
+            GenericContainer container = startContainerWithCommand(
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/wait/strategy/HttpWaitStrategyTest.java:49]: nit:
```suggestion
            GenericContainer<?> container = startContainerWithCommand(
```

**Review by bsideup — APPROVED** [review]
Approved with a tiny nit 🎉

**Pull request merged**